### PR TITLE
Added to context provider image to avoid multiple requests for the same image in UserPage

### PR DESCRIPTION
**Pull request by petromonka**
dev
## JIRA

* [Main JIRA ticket](https://jira.softserve.academy/secure/RapidBoard.jspa?rapidView=id)

## Code reviewers

- [ ] @github_username

### Second Level Review

- [ ] @github_username

## Summary of issue

ToDo

## Summary of change

ToDo

## Testing approach

ToDo

## CHECK LIST
- [ ]  СI passed
- [ ]  Сode coverage >=95%
- [ ]  PR is reviewed manually again (to make sure you have 100% ready code)
- [ ]  All reviewers agreed to merge the PR
- [ ]  I've checked new feature as logged in and logged out user if needed
- [ ]  PR meets all conventions

### CODE REVIEW

**Review by sharix4 — APPROVED** [review]

**Review by VolodymyrKostiv — APPROVED** [review]

File: src/pages/userPage/ActiveMembership/ActiveMembership.tsx [hunk 1]
```diff
@@ -58,30 +56,39 @@ const ActiveMembership = () => {
...
-      notificationLogic("error", error.message);
+    setAccessLevels(await activeMembershipApi.getAccessLevelById(userId));
+
+    await activeMembershipApi.getUserDates(userId).then((response) => {
+      response.dateEntry =
+        response.dateEntry === "0001-01-01T00:00:00" ? "" : response.dateEntry;
+      response.dateOath =
+        response.dateOath === "0001-01-01T00:00:00" ? "" : response.dateOath;
+      response.dateEnd =
+        response.dateEnd === "0001-01-01T00:00:00" ? "" : response.dateEnd;
```
Comment (OleksandrBilan) [inline src/pages/userPage/ActiveMembership/ActiveMembership.tsx:78]: you have the magic string "0001-01-01T00:00:00", which repeats 3 times. It would be better to create the const variable for this string

File: src/pages/userPage/personalData/PersonalData.tsx [hunk 2]
```diff
@@ -54,22 +57,27 @@ export default function ({
     setActiveUserRoles(userRoles);
     let currentUserId = UserApi.getActiveUserId();
     setActiveUserId(userId);
-    let userProfile = await UserApi.getActiveUserProfile();
-    setActiveUserProfile(userProfile);
+    let userProfile1 = await UserApi.getActiveUserProfile();
+    setActiveUserProfile(userProfile1);
```
Comment (OleksandrBilan) [inline src/pages/userPage/personalData/PersonalData.tsx:61]: why did you change the name to 'userProfile1' ?

File: src/pages/userPage/personalData/PersonalData.tsx [hunk 3]
```diff
@@ -54,22 +57,27 @@ export default function ({
...
+        UserApi.getImage(response.data?.user.imagePath ).then((response: { data: any }) => {
+          setImageBase64(response.data);
+        });
       })
       .catch((error) => {
         notificationLogic("error", error.message);
       });
+     
       setLoading(true);
+
```
Comment (christykobyletska) [inline src/pages/userPage/personalData/PersonalData.tsx:75]: do you need these empty lines?

File: src/pages/userPage/ActiveMembership/ActiveMembership.tsx (hunk 1 above)
Comment (petromonka) [reply src/pages/userPage/ActiveMembership/ActiveMembership.tsx:78]: Thank you for great advice! :)

File: src/pages/userPage/personalData/PersonalData.tsx (hunk 3 above)
Comment (petromonka) [reply src/pages/userPage/personalData/PersonalData.tsx]: I don't need them, so I removed them :)

**Review by christykobyletska — APPROVED** [review]
